OpenRA.Mods.Common/Activities/Air/HeliFly.cs

```diff
@@ -45,6 +46,18 @@ public static bool AdjustAltitude(Actor self, Helicopter helicopter, WDist targe
 			var dz = (targetAltitude.Length - altitude).Clamp(-delta, delta);
 			helicopter.SetPosition(self, helicopter.CenterPosition + new WVec(0, 0, dz));
 
+			if (dz > 0 && previousdz <= 0 && helicopter.Info.TakeoffSound != null)
```

Comment:
These checks will play the take off and landing sounds for normal altitude changes as helicopters move over bumpy terrain (once #7904 or a similar PR implements proper height handling for aircraft).